### PR TITLE
Improve Pipeline docs

### PR DESCRIPTION
**Pull request by bradhanks**
Part of #722

### CODE REVIEW

File: lib/membrane/pipeline.ex [hunk 1]
```diff
@@ -1,29 +1,25 @@
...
-  Pipelines are units that make it possible to instantiate, link and manage
-  elements and bins in convenient way (actually they should always be used inside
-  a pipeline). Linking pipeline children together enables them to pass data to one
-  another, and process it in different ways.
-
-  To create a pipeline, use the `__using__/1` macro and implement callbacks
-  of `Membrane.Pipeline` behaviour. For details on instantiating and linking
-  children, see `Membrane.ChildrenSpec`.
+  `Membrane.Pipeline` contains the callbacks and functions for constructing and supervising pipelines.
+  Pipelines facilitate the conveneient instantiation, linking, and management of elements and bins.\\
```
Comment (varsill) [inline lib/membrane/pipeline.ex:6]:
```suggestion
  Pipelines facilitate the convenient instantiation, linking, and management of elements and bins.\\
```

File: lib/membrane/pipeline.ex [hunk 2]
```diff
@@ -1,29 +1,25 @@
...
+  Linking pipeline children together enables them to pass and process data.
 
+  Use `use Membrane.Pipeline` and its callbacks to implement the `Membrane.Pipeline` behaviour.\\
```
Comment (varsill) [inline lib/membrane/pipeline.ex:9]:
```suggestion
  To create a pipeline, use `use Membrane.Pipeline` and implement callbacks of `Membrane.Pipeline`  behaviour.
```

File: lib/membrane/pipeline.ex [hunk 3]
```diff
@@ -67,50 +63,58 @@ defmodule Membrane.Pipeline do
...
 
+  @typedoc """
+  Defines the return value of the `start/3` and `start_link/3`."
+  """
   @type on_start ::
           {:ok, supervisor_pid :: pid, pipeline_pid :: pid}
           | {:error, {:already_started, pid()} | term()}
 
+  @typedoc """
+  The Pipeline state.
```
Comment (varsill) [inline lib/membrane/pipeline.ex:88]: Just for the sake of consistency 
```suggestion
  The pipeline state.
```

Comment (varsill) [reply lib/membrane/pipeline.ex:88]: I believe it is also applicable to a few other places

**Review by varsill — COMMENTED** [review]
Thank you very much, I have left some minor comments :D

**Review by varsill — APPROVED** [review]
🥇

**Review by mat-hek — APPROVED** [review]

**Pull request merged**